### DISCUSSION TITLE
Bundlr Client and wagmi provider

### DISCUSSION THREAD

**Discussion by lsbyerley** [wevm/wagmi#776] [Q&A]
Has anyone successfully used the provider from wagmi to initialize a [bundlr instance](https://docs.bundlr.network/docs/client/web)? When i try to pass the provider from wagmi to the `WebBundlr` class, I get the following error:

`TypeError: this.wallet.getSigner is not a function`

![Screen Shot 2022-07-28 at 9 16 40 AM](https://user-images.githubusercontent.com/3066258/181514494-08f225e0-f078-4a51-9dfc-771e455ba436.png)

```js
import { WebBundlr } from '@bundlr-network/client';
import { useProvider } from 'wagmi';

const provider = useProvider();

const initialiseBundlr = async () => {
  const bundlr = new WebBundlr(
    'https://node1.bundlr.network',
    'matic',
    provider
  );
  await bundlr.ready(); // ERROR HERE

  return bundlr;
};
```

**Comment by pavel-lens** [comment]
Did you see this article?
https://ropats16.hashnode.dev/build-and-upload-a-fully-decentralised-app-using-bundlr

**Reply by lsbyerley** [reply]
Thanks for the link. I have not, but after looking into it further, the article is never instantiating the `WebBundlr` class on the client. It is using the cli to upload files.
